daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/EvCalendarLanguage.scala

```diff
@@ -96,3 +97,33 @@ class CalendarEv(
     cal
   }
 }
+
+class DateTimeFormatterEv(
+  calendarEv: CalendarEv,
+  localeEv: CalendarLanguageEv,
+  pattern: String,
+  eci: DPathElementCompileInfo,
+) extends Evaluatable[ThreadLocal[SimpleDateFormat]](eci)
+  with InfosetCachedEvaluatable[ThreadLocal[SimpleDateFormat]] {
+
+  override lazy val runtimeDependencies = Seq(localeEv)
+
+  override def compute(state: ParseOrUnparseState) = {
+    val calendar = calendarEv.evaluate(state)
+    val locale = localeEv.evaluate(state)
+
+    // As per ICU4J documentation, "Date formats are not synchronized. If multiple threads
+    // access a format concurrently, it must be synchronized externally." Rather than
+    // synchronzing, we create a ThreadLocal so each thread gets their own copy of the
+    // SimpleDateFormat
+    val dateFormatTL = new ThreadLocal[SimpleDateFormat] with Serializable {
+      override def initialValue = {
+        val formatter = new SimpleDateFormat(pattern, locale)
+        formatter.setCalendar(calendar)
+        formatter.setLenient(true) // TODO: should this use calendarCheckPolicy?
```

Comment:
I'm pretty sure this is wrong and makes it so calendar parsing is always lenient regardless of the `dfdl:calendarCheckPolicy` property, but this copies the existing logic from the old `tlDataFormatter` so at least doesn't cause any regressions and just fixes the one bug at hand.

I've attempted to make it correctly use calendarCheckPolicy, but it broke a number of tests, including some unexpected failures that I think need more investigation. Once I can create a reproducible test I'll open a ticket, but I think a fix might be best for the next release so we have time to deal with regressions.

Reply:
Agree we should fix the one problem, and defer the strict/lenient to next point release.